### PR TITLE
fix `Wrong html processing in the overridden document.write function` (close #1218)

### LINKED ISSUE #1218: Wrong html processing in the overridden document.write function
Example: 
```javascript
document.write('<script src="script.js">');
document.write('<\/script>');
```

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/utils/html.js [hunk 1]
```diff
@@ -221,8 +221,17 @@ export function processHtml (html, parentTag, prepareDom) {
         for (let i = 0; i < children.length; i++) {
             const el = children[i];
 
-            if (hasIsNotClosedFlag(el))
+            if (hasIsNotClosedFlag(el)) {
+                if (isScriptElement(el) && nativeMethods.hasAttribute.call(el, 'src')) {
+                    const storedTextContent = el.textContent;
```
Comment (miherlosev) [inline src/client/utils/html.js]: Need to investigate why do we need to preserve a `textContent`.
Add test for this case if it's need.

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/dom-processor/client-dom-adapter.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { getProxyUrl } from '../utils/url';
 import { isIE9 } from '../utils/browser';
 import * as domUtils from '../utils/dom';
 import fastApply from '../utils/fast-apply';
+import { hasIsNotClosedFlag } from '../sandbox/node/document/writer';
```
Comment (miherlosev) [inline src/client/dom-processor/client-dom-adapter.js]: After discussing with @VasilyStrelyaev  we decide use the `unclosed` word.
Accordingly, need to rename in the `DocumentWriter`:
```js
const IS_NOT_CLOSED_PROPERTY = 'hammerhead|is-not-closed-element';
hasIsNotClosedFlag
_setIsNotClosedFlag
...
const UNCLOSED_ELEMENT_FLAG = 'hammerhead|unclosed-element-flag';
setUnclosedElementFlag
hasUnclosedElementFlag
```

**Comment by churkin** [comment]
Can I review?

**Comment by miherlosev** [comment]
Yes

**Pull request merged**